### DISCUSSION TITLE
[PA3] wireshark [TCP CHECKSUM INCORRECT] in solution and SYN-ACK dropping with checksum 0xeeee

### DISCUSSION THREAD

**Discussion by Morivy42** [ANLAB-KAIST/KENSv3#228] [Project Q&A 2. Transport Layer (TCP)]
Hello 
I'm testing `test-kens-close-unreliable` and `test-kens-handshake-unreliable` but tests that `Test*_Accept_*` are fails.
The tests passed in WSL of my teammate but not in my M2 MacBook air with CLion.
With RANDOMSEED = 1234, tests passed but not others for me.
The Accept tests fails because SYN-ACK that my code sent is dropped. I can't figure out why. 

1. When I check TCP checksum of solution using Wireshark, the SYN-ACK packet is [TCP CHECKSUM INCORRECT].
Is this just problem of my environment?
<img width="791" alt="스크린샷 2024-05-07 00 47 00" src="https://github.com/ANLAB-KAIST/KENSv3/assets/86469551/d38cdec5-1373-4105-902c-329bd5dc3467">

also here's log for `TestEnv_Any.TestAccept_AfterAccept` with `TCP_LOG` flag in `E_NetworkLog.cpp`
![스크린샷 2024-05-07 00 55 57](https://github.com/ANLAB-KAIST/KENSv3/assets/86469551/f05fbef5-03cf-422f-8e0a-1e672ddea55a)

2. SYN-ACK keeps dropping even though the checksum is same with solution by 0xeeee. What is the correct checksum?
Checksum is same with solution as 0xeeee. Right is solution's SYN-ACK in `TestAccept_AfterAccept`
![스크린샷 2024-05-07 00 58 03](https://github.com/ANLAB-KAIST/KENSv3/assets/86469551/b6df5888-658d-45a4-aff7-d27e03da0fab)

`#define UNRELIABLE` is already commented.

I think the difference of SYN-ACK between my code and solution is here. but I can't find where this part is in packet.
![스크린샷 2024-05-07 00 41 17](https://github.com/ANLAB-KAIST/KENSv3/assets/86469551/5c80712d-41fb-4d62-a256-96acf3a510bc)

They work fine with reliable tests.

Thank you.

**Comment by seo-rii** [comment]
FYI) Since random generator is system-dependent, same random_seed may cause other result in different OS.

**Reply by seo-rii** [reply]
https://github.com/ANLAB-KAIST/KENSv3/discussions/221

**Reply by Morivy42** [reply]
Thx. As in #79, I'm trying to fix for both environment. Also, I think my code is working correctly for few random_seed now.
